File: solutions/csharp/0151-reverse-words-in-a-string.cs
public class Solution {
    public string ReverseWords(string s) {
        string[] words = s.Trim().Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
        Array.Reverse(words);
        
        StringBuilder result = new StringBuilder();
        
        for (int i = 0; i < words.Length; i++) {
            result.Append(words[i]);
            if (i < words.Length - 1) result.Append(" ");
        }
        
        return result.ToString();
    }
}

File: solutions/cpp/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        std::istringstream iss(s);
        std::vector<std::string> words;
        std::string word;
        
        while (iss >> word) {
            words.push_back(word);
        }
        
        std::reverse(words.begin(), words.end());
        
        std::string result;
        for (int i = 0; i < words.size(); ++i) {
            result += words[i];
            if (i < words.size() - 1) result += ' ';
        }
        
        return result;        
    }
};
